site/source/tutorials/accessibility.md

```diff
@@ -0,0 +1,96 @@
+
+The `label` property on both form widgets and the `Label` widget accepts either a string or an [options object](https://github.com/dojo/widgets/blob/master/src/label/Label.ts#L13-17) that allows customization of where the label is placed (before or after the input), and whether it is visually hidden. Visually hiding a label is done by using screen reader-accessible CSS styles to hide text content, and is recommended for use with form controls that have no visible label.
+
+A `<label>` element with hidden text was chosen over the `aria-label` attribute for invisible labels due to [still-inconsistent](https://www.powermapper.com/tests/screen-readers/labelling/input-text-aria-label/) screen reader support for the latter.
+
+### Styling
+
+The base Dojo 2 theme meets [WCAG AA](https://www.w3.org/TR/WCAG20/#conformance-reqs) color contrast guidelines including dedicated styling rules to help users know which node is currently focused, but the developer must check the final styles as changes to font size or background could affect readability.
+
+In addition to themes, there is a set of base styles provided by `@dojo/widgets` containing basic utility classes like the screen reader-accessible `visuallyHidden` style. To use it, import `baseCss` separately like this:
+
+```ts
+import * as css from ‘path/to/your/css’;
+import * as baseCss from ‘@dojo/widgets/common/styles/base.m.css’;
+
+@theme(css)
+class MyWidget extends WidgetBase {
+    render() {
+        return v(‘div’, {
+            classes: this.classes().fixed(baseCss.visuallyHidden)
+        }, [ ‘Screen reader instructions’ ]);
+    }
+}
+```
+
+This CSS class will hide a widget visually while allowing it to remain visible to screen readers.
+
+## Writing custom widgets
+
+All DOM attributes can be set with `VirtualDomProperties` so, apart from managing focus, no extra tools are required to create highly accessible widgets. For example, the following code would create an accessible popup button:
+
+```ts
+v('button', {
+    key: 'buttonKey',
+    'aria-controls': 'popupId',
+    'aria-expanded': String(this.properties.expanded),
+    'aria-haspopup': 'true',
+    id: 'buttonId',
+    onclick: this._togglePopup
+}, [ 'Click to open popup' ])
+```
+
+Since this example uses a `<button>` element, it requires neither an explicit `tabindex` nor an `onkeydown` event to work properly with a keyboard. Wherever possible, Dojo 2 takes advantage of built-in accessibility by using native elements. However, while native functionality is the most straightforward path to good accessibility, there are times when it is not possible, such as when no built-in element gives the desired functionality.
+
+For those times, this checklist is a good starting point to ensure that a widget is accessible:
```

Comment:
"For those times," -> "For those times when native functionality is insufficient,"